code/game/machinery/deployable.dm

```diff
@@ -210,6 +223,202 @@
+	///The core shield that protects the generator
+	var/obj/structure/barricade/dropwall/core_shield = null
+
+/obj/structure/dropwall_generator/Initialize(mapload, direction, uptime)
+	. = ..()
+	if(direction)
+		deploy(direction, uptime)
+
+/obj/structure/dropwall_generator/proc/deploy(direction, uptime)
+	anchored = TRUE
+	protected = TRUE
+	addtimer(CALLBACK(src, .proc/power_out), uptime)
+	while(uptime >= 10)
+		addtimer(CALLBACK(src, .proc/annoying_overlay_proc, uptime / 10), (10 * (12 - uptime / 10))) // instantly, 1 second, 2 seconds, ect. Would put in SECONDS here but it shits itself
+		uptime -= 10
+
+	switch(direction)
+		if(NORTH)
+			connected_shields += new /obj/structure/barricade/dropwall(get_turf(loc), src, TRUE, NORTH)
+			for(var/i in connected_shields)
+				core_shield = i
+
+			var/target_turf = get_step(src, EAST)
+			if(!(is_blocked_turf(target_turf)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf, src, FALSE, NORTH, EAST)
+
+			var/target_turf2 = get_step(src, WEST)
+			if(!(is_blocked_turf(target_turf2)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf2, src, FALSE, NORTH, WEST)
+		if(EAST)
+			connected_shields += new /obj/structure/barricade/dropwall(get_turf(loc), src, TRUE, EAST)
+			for(var/i in connected_shields)
+				core_shield = i
+
+			var/target_turf = get_step(src, NORTH)
+			if(!(is_blocked_turf(target_turf)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf, src, FALSE, NORTH, EAST)
+
+			var/target_turf2 = get_step(src, SOUTH)
+			if(!(is_blocked_turf(target_turf2)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf2, src, FALSE, SOUTH, EAST)
+		if(SOUTH)
+			connected_shields += new /obj/structure/barricade/dropwall(get_turf(loc), src, TRUE, SOUTH)
+			for(var/i in connected_shields)
+				core_shield = i
+
+			var/target_turf = get_step(src, EAST)
+			if(!(is_blocked_turf(target_turf)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf, src, FALSE, SOUTH, EAST)
+
+			var/target_turf2 = get_step(src, WEST)
+			if(!(is_blocked_turf(target_turf2)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf2, src, FALSE, SOUTH, WEST)
+		if(WEST)
+			connected_shields += new /obj/structure/barricade/dropwall(get_turf(loc), src, TRUE, WEST)
+			for(var/i in connected_shields)
+				core_shield = i
+
+			var/target_turf = get_step(src, NORTH)
+			if(!(is_blocked_turf(target_turf)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf, src, FALSE, NORTH, WEST)
+
+			var/target_turf2 = get_step(src, SOUTH)
+			if(!(is_blocked_turf(target_turf2)))
+				connected_shields += new /obj/structure/barricade/dropwall(target_turf2, src, FALSE, SOUTH, WEST)
+
+
+/obj/structure/dropwall_generator/attacked_by(obj/item/I, mob/living/user) //No, you can not just go up to the generator and whack it. Central shield needs to go down first.
```

Comment:
Discussed on discord, "...using attacked_by works better than attackby. Attackby it where the animation and click cooldown is used. If I override the attackby, the animation does not play and there is no cooldown. However, attacked_by comes after, and is where the object takes damage and sends the messages. Thus, by changing attacked by, I can change the damage to the shield instead of the shield generator, without messing with the animation or the click delay"